### PR TITLE
Deploy router controller resource from bundle 

### PR DESCRIPTION
**Pull request by praveenkumar**
As part of https://github.com/crc-org/snc/pull/738, we put
router-controller resource as part of bundle and from crc side
we don't need to generate this resource.

### CODE REVIEW

**Comment by praveenkumar** [comment]
/hold

Till https://github.com/crc-org/snc/pull/738 merge and new bundle available to consume.

File: pkg/crc/machine/start.go
Comment (cfergeau) [inline pkg/crc/machine/start.go]: Can this go to `/usr/share/crc/routes_controller.yaml` or such?

**Review by cfergeau — COMMENTED** [review]
After this change it looks like crc won't be able to start older bundles?

File: pkg/crc/machine/start.go
Comment (praveenkumar) [reply pkg/crc/machine/start.go]: @cfergeau any specific reason because we are storing the `kubeconfig` in the same location so didn't want to change it.

Comment (cfergeau) [reply pkg/crc/machine/start.go]: I know there are other files there, I made a comment at the time, but I should have pushed stronger :) 
/opt is an odd place to put crc-specific files, especially without subdirectories.

Comment (praveenkumar) [reply pkg/crc/machine/start.go]: @cfergeau updated.

**Comment by praveenkumar** [comment]
>  After this change it looks like crc won't be able to start older bundles?

@crc-org/crc-team  Older bundles will not work after this PR merge. Question is do we want to support or need to have this backward compatibility for some releases?

File: pkg/crc/machine/start.go
Comment (praveenkumar) [reply pkg/crc/machine/start.go]: https://github.com/crc-org/snc/pull/738#issuecomment-1591023529

**Comment by cfergeau** [comment]
> > After this change it looks like crc won't be able to start older bundles?
> 
> @crc-org/crc-team Older bundles will not work after this PR merge. Question is do we want to support or need to have this backward compatibility for some releases?

Imo yes, at least 4.13.x bundles, and maybe even 4.12.x. But this is a team decision to make.

**Comment by gbraad** [comment]
We are allowed to cut off support, but as long as we inform the user of this choice.
Since we do not know how many 'custom' bundles are used, this is necessary to be reported

**Comment by cfergeau** [comment]
> We are allowed to cut off support, but as long as we inform the user of this choice.

"allowed" is a thing, but what do we _want_ as a team? break backwards compat any time we need to with a warning? document that a given CRC release will support all 4.13.x bundles if it was released with 4.13.x? Go further than that and support both 4.12 and 4.13 bundles if it was released with 4.13?

File: pkg/crc/machine/start.go
Comment (gbraad) [reply pkg/crc/machine/start.go]: `/opt/crc` makes perfectly sense to me. In my opinion, locations under `/usr` should only be populated by packages.

**Comment by gbraad** [comment]
It is easy to check for the file existence in `/opt/crc` to see if we need to generate the json file instead

**Comment by gbraad** [comment]
> Go further than that and support both 4.12 and 4.13 bundles if it was released with 4.13?

Also, let's not assume anything. We could easily find the information if someone uses an older bundle with a newer version of crc, or a custom bundle. Telemetry... anyone?

Besides, we only 'support' N-1. So anything below OpenShift 4.13 that was part of [2.20.0-4.13.0](https://github.com/crc-org/crc/releases/tag/v2.20.0) by the time we release this (possibly) as part of 2.21, will not receive attention.

**Comment by praveenkumar** [comment]
/unhold

As of now I didn't remove the extra  code to use the older bundle so it now shouldn't break but in future release after 2.24 I would remove that extra code.

File: pkg/crc/machine/start.go
Comment (cfergeau) [inline pkg/crc/machine/start.go:933]: Why 2.24? As a prerequisite for removing this, I'd first start with having a clear policy wrt bundles, and a clear error message that the bundle is not supported, not a cryptic failure.

**Review by cfergeau — APPROVED** [review]

File: pkg/crc/machine/start.go
Comment (cfergeau) [reply pkg/crc/machine/start.go]: `/usr/share` is read-only in RHCOS, but `/usr/share/local/crc/` is also a 'good' location, not owned by packages.

File: pkg/crc/machine/start.go
Comment (praveenkumar) [reply pkg/crc/machine/start.go:933]: I can see you already created #3730 for that.

**Pull request merged**
